### ISSUE TITLE
Panic while performing terraform plan

### ISSUE THREAD

**Issue by jpadhye** [DataDog/terraform-provider-datadog#633]
### Terraform Version
```
$ /opt/terraform/terraform -v
Terraform v0.11.14
+ provider.datadog v2.12.1

Your version of Terraform is out of date! The latest version
is 0.13.0. You can update by downloading from www.terraform.io/downloads.html
```
### Resources
```
data.terraform_remote_state.evt_infra: Refreshing state...
data.terraform_remote_state.evt_global: Refreshing state...
datadog_dashboard.ordered_dashboard: Refreshing state... (ID: b92-5xu-4tx)

Error: Error refreshing state: 1 error occurred:
	* datadog_dashboard.ordered_dashboard: 1 error occurred:
	* datadog_dashboard.ordered_dashboard: datadog_dashboard.ordered_dashboard: unexpected EOF
```
### Panic Output
```
-----------------------------------------------------
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 2020/08/13 13:37:39
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: HTTP/1.1 200 OK
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Transfer-Encoding: chunked
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Cache-Control: no-cache
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Connection: keep-alive
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Content-Security-Policy: frame-ancestors 'self'; report-uri https://api.datadoghq.com/csp-report
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Content-Type: application/json
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Date: Thu, 13 Aug 2020 20:37:39 GMT
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Dd-Pool: dogweb
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Pragma: no-cache
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Set-Cookie: DD-PSHARD=162; Max-Age=604800; Path=/; expires=Thu, 20-Aug-2020 20:37:39 GMT; secure; HttpOnly
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Strict-Transport-Security: max-age=15724800;
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: Vary: Accept-Encoding
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: X-Content-Type-Options: nosniff
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: X-Dd-Debug: dHGWGBt164MQ3c958Df9fAGdV1KjD7IUXsiVb89HiZaBnA9hzGJWLUoOh9YtWL02
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: X-Dd-Version: 35.2876286
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: X-Frame-Options: SAMEORIGIN
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 
2020-08-13T13:37:39.422-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 9201
2020-08-13T13:37:39.423-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 0
2020-08-13T13:37:39.423-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 
2020-08-13T13:37:39.423-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: panic: runtime error: invalid memory address or nil pointer dereference
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1bddcf1]
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: goroutine 147 [running]:
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: github.com/terraform-providers/terraform-provider-datadog/datadog.buildTerraformTemplateVariables(0xc00060ba40, 0x1fa2ab9)
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:296 +0x291
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogDashboardRead(0xc0004f7650, 0x1d14040, 0xc00028a740, 0xc0004f7650, 0x0)
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:152 +0x463
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Refresh(0xc00030a6c0, 0xc0003d8050, 0x1d14040, 0xc00028a740, 0xc00019a788, 0x100c101, 0xc000722330)
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 	github.com/hashicorp/terraform-plugin-sdk@v1.10.0/helper/schema/resource.go:516 +0x16f
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Refresh(0xc00035a000, 0xc0000a2580, 0xc0003d8050, 0xc000722330, 0xc000526000, 0x5bb)
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 	github.com/hashicorp/terraform-plugin-sdk@v1.10.0/helper/schema/provider.go:333 +0x8f
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: github.com/hashicorp/terraform-plugin-sdk/plugin.(*ResourceProviderServer).Refresh(0xc0006ee180, 0xc000980000, 0xc000980250, 0x0, 0x0)
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 	github.com/hashicorp/terraform-plugin-sdk@v1.10.0/plugin/resource_provider.go:552 +0x4e
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: reflect.Value.call(0xc000045260, 0xc0000a8120, 0x13, 0x1fa0066, 0x4, 0xc000798f08, 0x3, 0x3, 0x0, 0x0, ...)
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 	reflect/value.go:460 +0x8ab
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: reflect.Value.Call(0xc000045260, 0xc0000a8120, 0x13, 0xc000795708, 0x3, 0x3, 0x0, 0x0, 0x0)
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 	reflect/value.go:321 +0xb4
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: net/rpc.(*service).call(0xc00018e1c0, 0xc0003620a0, 0xc0000a60b0, 0xc0000a60c0, 0xc00035a500, 0xc0006ee940, 0x1d135c0, 0xc000980000, 0x16, 0x1d13600, ...)
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 	net/rpc/server.go:377 +0x17f
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: created by net/rpc.(*Server).ServeCodec
2020-08-13T13:37:39.456-0700 [DEBUG] plugin.terraform-provider-datadog_v2.12.1: 	net/rpc/server.go:474 +0x42b
2020/08/13 13:37:39 [ERROR] root: eval: *terraform.EvalRefresh, err: datadog_dashboard.ordered_dashboard: unexpected EOF
2020/08/13 13:37:39 [ERROR] root: eval: *terraform.EvalSequence, err: datadog_dashboard.ordered_dashboard: unexpected EOF
2020/08/13 13:37:39 [TRACE] [walkRefresh] Exiting eval tree: datadog_dashboard.ordered_dashboard
2020/08/13 13:37:39 [TRACE] dag/walk: upstream errored, not walking "provider.datadog (close)"
2020-08-13T13:37:39.458-0700 [DEBUG] plugin: plugin process exited: path=/deployments/production/dataDashBoard/.terraform/plugins/darwin_amd64/terraform-provider-datadog_v2.12.1
-----------------------------------------------------
```
### Steps to Reproduce
terraform plan

**Comment by therve** [comment]
Thanks for the report, can you share your terraform file if possible?

**Comment by therve** [comment]
It looks like the issue fixed by #630 . We'll release soon with that fix included.
